### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by Skunkchain**
updatooor

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,23 +2,35 @@
 
 ## Overview
 
-Abacus is a cross-chain communication system. It handles passing raw buffers
-between blockchains cheaply, and with minimal fuss. Like IBC and other
-cross-chain communication systems, Abacus creates channels between chains, and
-then passes its messages over the channel. Once a channel is established, any
-application on the chain can use it to send messages to any other chain.
+Abacus is a interchain messaging protocol. It handles passing generalized messages
```
Comment (nambrot) [inline README.md:5]: an

File: README.md [hunk 2]
```diff
@@ -2,23 +2,35 @@
...
+Abacus creates channels between chains. In that respect, Abacus is a network between blockchains,
+and then passes messages between applications over these channels. Channels can be used 
+by numerous applications on any source or destination chain. Abacus is implemented wholly in
+smart contracts, and thus can be implemented in any turing complete smart contract chain.
 
-Compared to IBC and PoS light client based cross-chain communication, Abacus
-has weaker security guarantees. However, Abacus may be implemented on any smart
-contract chain, with no bespoke light client engineering. Because it does not run
-a light client, Abacus does not spend extra gas verifying remote chain block headers.
+Unfortunately, while Abacus might feel like magic, it isn't actually magic. Something's got to give.
```
Comment (nambrot) [inline README.md:12]: This feels a bit more casual then I would expect in a technical repo

File: README.md [hunk 3]
```diff
@@ -104,47 +116,44 @@ cd rust
...
-We present Abacus — a system for sending messages between consensus systems
-without paying header validation costs by creating the illusion of cross-chain
-communication. Similar to an atomic swap, Abacus uses non-global protocol
-validation to simulate cross-chain communication. Abacus can carry arbitrary
-messages (raw byte vectors), uses a single-producer multi-consumer model, and
-has protocol overhead sublinear in the number of messages being sent.
+Abacus is a protocol that enables interchain applications by passing messages, any raw bytes, 
+between smart contract blockchains. It does so without bearing the expensive costs involved
+with header verification. Notably, Abacus has economies of scale in messaging costs, and
+brings a new security model to the fore in its Proof of Stake combined with our very own;
```
Comment (nambrot) [inline README.md:122]: TIL to bring something to the fore

Comment (nambrot) [reply README.md:122]: odd semi-colon on this line

File: README.md [hunk 4]
```diff
@@ -104,47 +116,44 @@ cd rust
...
 ## Summary
 
-Abacus is a new strategy for simulating cross-chain communication without
-validating headers. The goal is to create a single short piece of state (a
-32-byte hash) that can be updated regularly. This hash represents a merkle tree
-containing a set of cross-chain messages being sent by a single chain (the
-"outbox" chain for the Abacus system). Contracts on the outbox chain can submit
-messages, which are put into a merkle tree (the "message tree"). The message
-tree's root may be transferred to any number of "inbox" chains.
+Abacus is a protocol for interchain communication. The protocol's goal is to use generalized messaging between smart contract blockchains to enable interchain applications.
```
Comment (nambrot) [inline README.md:144]: I just noticed this now, but "What is abacus" seems to repeated 3 times in this readme, we should probably consolidate

**Review by nambrot — COMMENTED** [review]
Looks good, though overall, I would maybe tend to keep this kind of information in the docs, significantly have this kind of high level description of Abacus be more concise and focus more on explaining this code base, how it is organized, and how to build/test/develop in it

File: README.md (hunk 2 above)
Comment (Skunkchain) [reply README.md:12]: thats the goal ;)

**Review by nambrot — APPROVED** [review]

**Pull request merged**